checks.d/openstack.py

```diff
@@ -501,10 +616,33 @@ def check(self, instance):
             self.log.debug("Auth Token: %s", self._auth_token)
 
             try:
-                hyp_stats = self.get_hypervisor_stats()
-                self.nova_stats_to_metrics(hyp_stats)
+                if self._local_only:
+                    # The new default: restrict monitoring to this (host, hypervisor, project)
+                    # and it's guest servers
+
+                    hyp = self.get_local_hypervisor()
+                    project = self.get_local_project()
+
+                    aggregate = self.get_aggregates_for_local_host()
+
+                    # Restrict monitoring to hyp and servers, don't do anything else
+                    servers = self.get_servers_managed_by_hypervisor()
+                    host_tags = self._get_tags_for_host()
+
+                    for sid in servers:
+                        server_tags = host_tags + ['host:%s' % sid]
+                        self.get_stats_for_single_server(sid, tags=server_tags)
```

Comment:
Need to try/except this block in case there is an error getting stats for a single server. (Not sure why but one of our instances is failing this call. I think it is in a corrupt state). Same for in the `get_server_stats` method.